library/dellemc/idrac/_dellemc_install_firmware.py

```diff
@@ -19,98 +18,210 @@
-    idrac_pwd:
-        required: True
-        description: iDRAC user password.
-    idrac_port:
-        required: False
-        description: iDRAC port.
-        default: 443
-    share_name:
-        required: True
-        description: CIFS or NFS Network share.
-    share_user:
-        required: False
-        description: Network share user in the format 'user@domain' or 'domain\\user' if user is
-            part of a domain else 'user'. This option is mandatory for CIFS Network Share.
-    share_pwd:
-        required: False
-        description: Network share user password. This option is mandatory for CIFS Network Share.
-    share_mnt:
-        required: True
-        description: Local mount path of the network share with read-write permission for ansible user.
-            This option is mandatory for Network Share.
-    reboot:
-        required: False
-        description: Whether to reboots after applying the updates or not.
-        default: False
-        type: bool
-    job_wait:
-        required:  True
-        description: Whether to wait for job completion or not.
-        type: bool
-        default: True
-    catalog_file_name:
-        required: False
-        description: Catalog file name relative to the I(share_name).
-        type: str
-        default: 'Catalog.xml'
+  idrac_ip:
+    required: True
+    description:
+      - iDRAC IP Address
+    type: 'str'
+  idrac_user:
+    required: True
+    description:
+      - iDRAC user name
+    type: 'str'
+  idrac_pwd:
+    required: True
+    description:
+      - iDRAC user password
+    type: 'str'
+  idrac_port:
+    required: False
+    description:
+      - iDRAC port
+    default: 443
+    type: 'int'
+  share_name:
+    required: True
+    description:
+      - Network share (CIFS, NFS, HTTP, HTTPS, FTP) containing the Catalog file
+        and Update Packages (DUPs)
+    type: 'str'
+  share_user:
+    required: False
+    description:
+      - Network share user in the format 'user@domain' or 'domain\\user' if
+        user is part of a domain else 'user'. This option is mandatory if
+        I(share_name) is a CIFS share.
+    type: 'str'
+  share_pwd:
+    required: False
+    description:
+      - Network share user password
+    type: 'str'
+  share_mnt:
+    required: False
+    description:
+      - Local mount path on the ansible controller machine for the remote
+        network share (CIFS, NFS) provided in I(share_name). This is not
+        applicable for HTTP, HTTPS and FTP share.
+      - This option is mandatory only when using firmware update from a network
+        repository using Server Configuration Profiles (SCP).
+      - SCP based firmware update is only supported for iDRAC firmware
+        version >=3.00.00.00.
+    type: 'path'
+  catalog_file_name:
+    required: False
+    description:
+      - Catalog file name relative to the I(share_name)
+    default: 'Catalog.xml'
+    type: 'str'
+  apply_update:
```

Comment:
When and how the firmware will get updated in case this `'apply_update'` flag is set False?

How consumer can ensure the firmware get applied in case the flag is set 'False', Describe the steps involved for the same


Customer want to stage the firmware and Catalog and apply it at later point of time he can schedule the playbook run. When we built in scheduling capability with in the command, will make it difficult for seamless integration with AWX and Tower?